nvtabular/ops.py

```diff
@@ -126,6 +126,9 @@ def assemble_new_df(self, origin_gdf, new_gdf, target_columns):
         if self.replace and self.preprocessing and target_columns:
             if new_gdf.shape[0] < origin_gdf.shape[0]:
                 return new_gdf
+            # handle case when the dropna operator doesn't drop any rows
```

Comment:
The 'Dropna' operator is returning all the columns - and filtering out based on whether the target_columns contain nulls or not (I'm not sure this is correct behaviour, but it is whats happening). When it didn't filter out any rows, it returned more columns than target_columns causing a size mismatch here ```gdf[target_columns] = new_gdf ``` (when it did filter out rows we just returned new_gdf directly). This problem is triggered with these changes because we aren't passing columns to the GPUDatasetIterator class anymore - but I believe could have happened anyways

Reply:
@benfred yes, `dropna()` is supposed to drop the rows where there are nulls based on the target columns. so, the retuned new_gdf, can have less rows than the original gdf, but should not have less columns, we are not dropping columns here. for this we can use `drop` op.  Not sure, I understand the error `When it didn't filter out any rows, it returned more columns than target_columns `.  I tested that situation, but I was not getting more columns than the original. It should not be. Did I miss a logic when creating the op? Let me know how I can help.

Reply:
@rynak Before this change we were iterating over a subset of columns available , and target_columns was every column in the dataframe. Now there are an extra 2 columns - and we got a dimensionality mismatch on line 133 (```origin_gdf[target_columns] = new_gdf```) here without this check.  After talking this over with Julio, I changed to override apply_op instead of changing this function in the base class to handle this